test/object-store/c_api/c_api.cpp

```diff
@@ -3962,6 +3962,163 @@ struct Userdata {
 };
 
 #if REALM_ENABLE_SYNC
+
+TEST_CASE("C API - client reset", "[c_api][client-reset]") {
+    reset_utils::Partition partition{"realm_id", random_string(20)};
+    Property partition_prop = {partition.property_name, PropertyType::String | PropertyType::Nullable};
+    Schema schema{
+        {"object",
+         {
+             {"_id", PropertyType::ObjectId, Property::IsPrimary{true}},
+             {"value", PropertyType::Int},
+             partition_prop,
+         }},
+    };
+
+    std::string base_url = get_base_url();
+    REQUIRE(!base_url.empty());
+    auto server_app_config = minimal_app_config(base_url, "c_api_client_reset_tests", schema);
+    server_app_config.partition_key = partition_prop;
+    TestAppSession test_app_session(create_app(server_app_config));
+
+    auto app = test_app_session.app();
+    auto get_valid_config = [&]() -> SyncTestFile {
+        create_user_and_log_in(app);
+        return SyncTestFile(app->current_user(), partition.value, schema);
+    };
+    SyncTestFile local_config = get_valid_config();
```

Comment:
I tried to create these sync test files via the normal C-API but for some reason something is not working, the tests will hang.. I will review this tomorrow..

Reply:
Although doing everything in the c-api would make a more comprehensive integration test, it seems fine to me to use a SyncTestFile instance here. It is just the test framework and not the code we actually want to test.